runtime/cmd/check/main.go

```diff
@@ -184,14 +190,18 @@ func runPath(path string, bench bool) (res result, succeeded bool) {
 			}
 		}()
 
-		program, must = cmd.PrepareProgram(code, path, codes)
+		program, must = cmd.PrepareProgram(code, location, codes)
 
-		checker, _ = cmd.PrepareChecker(program, path, codes, must)
+		checker, _ = cmd.PrepareChecker(program, location, codes, must)
 
 		err = checker.Check()
 		if err != nil {
 			var builder strings.Builder
-			cmd.PrettyPrintError(&builder, err, path, codes)
+			printErr := pretty.NewErrorPrettyPrinter(&builder, useColor).
+				PrettyPrintError(err, location, codes)
+			if printErr != nil {
```

Comment:
all currant usages of `PrettyPrintError` panic if the function returns an error... should that perhaps be the responsibility of `PrettyPrintError`?

Reply:
I think letting callers decide if a pretty printing error is fatal or not is better than always making it fatal. You're right that in all current cases where the pretty printer is used, an error is fatal, but it doesn't need to be in future uses.